static/js/apps/visualization/app_context.tsx

```diff
@@ -76,18 +77,21 @@ export function AppContextProvider(
   props: AppContextProviderPropType
 ): JSX.Element {
   const [places, setPlaces] = useState([]);
-  const [enclosedPlaceType, setEnclosedPlaceType] = useState(
-    getParamValue(URL_PARAMS.ENCLOSED_PLACE_TYPE) || ""
-  );
+  const [enclosedPlaceType, setEnclosedPlaceType] = useState("");
   const [statVars, setStatVars] = useState([]);
   const [visType, setVisType] = useState(
     getParamValue(URL_PARAMS.VIS_TYPE) || ORDERED_VIS_TYPE[0].valueOf()
   );
   const [childPlaceTypes, setChildPlaceTypes] = useState(null);
   const [samplePlaces, setSamplePlaces] = useState([]);
-  const [isInitializing, setIsInitializing] = useState(true);
+  const [isContextLoading, setIsContextLoading] = useState(true);
   const prevSamplePlaces = useRef(samplePlaces);
   const prevStatVars = useRef(statVars);
+  // List of booleans to track whether or not hash should be updated when
+  // places, enclosedPlaceType, statVars, or visType changes. Hash should be
+  // updated for user selections and NOT for selections that are auto-magically
+  // made
+  const shouldUpdateHash = useRef([]);
```

Comment:
does the order of this list corresponding to place, statVars etc..? If so, is a map better?